fix(hooks): Reorder the UI elements so that `PreToolUse` appears above tool

candieduniverse opened the pull request:
### Related Issue

**Issue:** n/a

### Description

This PR reorders the `PreToolUse` hook UI block to appear above the "Cline wants to" tool use message in the webview.  See screenshots for a visual explanation of how this improves the user experience in the extension.


### Test Procedure

I tested this manually by enabling hooks and configuring a `PreToolUse` hook and then I ran a simple task to verify that the ordering changes to what we want it to be.  Ran this against `main` branch for the "before" and then ran this against this PR branch for the "after".

### Type of Change

-   [x] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [ ] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [ ] 🏃 Workflow Changes

### Pre-flight Checklist

-   [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [x] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [ ] I have created a changeset using `npm run changeset` (required for user-facing changes)
-   [x] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### Screenshots

#### Before
<img width="699" height="372" alt="Screenshot 2025-11-13 at 3 50 49 PM" src="https://github.com/user-attachments/assets/0bb0d247-f239-4edf-a9ea-4a7e84e0db6d" />


#### After
<img width="517" height="381" alt="Screenshot 2025-11-13 at 3 47 29 PM" src="https://github.com/user-attachments/assets/26cbc5c6-b373-4752-a46e-ec2c9c1051e2" />


### Additional Notes

Thank you for talking about this yesterday and coming up with a simpler solution than the direction I'd gone down, @abeatrix!  That was very helpful.

abeatrix commented on src/core/hooks/hook-executor.ts:
```diff
@@ -224,3 +230,47 @@ async function updateHookMessage(
+ * 4. Re-add the tool message at the end (after hook messages)
+ */
+async function reorderHookAndToolMessages(messageStateHandler: MessageStateHandler): Promise<void> {
+	const { findLastIndex } = await import("@shared/array")
```
Curious the preference of doing dynamic imports for findLastIndex

candieduniverse replied:
Removing `findLastIndex`.

abeatrix reviewed (dismissed):
Noticed some unexpected behavior when testing the hooks: the pre tool use is getting executed before I can approve a tool use, and the pretool use hook shows up after the "Cline wants to execute this command" block, even when the command was not executed / approved:

<img width="507" height="470" alt="image" src="https://github.com/user-attachments/assets/85aa1b15-6474-4bf8-a4bf-4dde7800137d" />

candieduniverse commented:
I've addressed the code review feedback and quirky UI issues from yesterday.  I've verified the UI behavior and things look to me.  This PR is ready again for review.
